pkg/ipam/poolallocator/allocator.go

```diff
@@ -152,19 +153,55 @@ func (a *IPPoolAllocator) appendPoolUsage(ipPool *v1alpha2.IPPool, ip net.IP, st
 
 }
 
-// Update pool status to delete released IP
+func (a *IPPoolAllocator) updatePoolUsage(ipPool *v1alpha2.IPPool, ip net.IP, state v1alpha2.IPAddressPhase, owner v1alpha2.IPAddressOwner) error {
+	newPool := ipPool.DeepCopy()
+	ipString := ip.String()
+	found := false
+
+	for i, ipAddress := range newPool.Status.IPAddresses {
+		if ipAddress.IPAddress == ipString {
+			newPool.Status.IPAddresses[i].Phase = state
+			newPool.Status.IPAddresses[i].Owner = owner
+			found = true
+			break
+		}
+	}
+	if !found {
+		return fmt.Errorf("ip %s usage not found in pool %s", ipString, newPool.Name)
+	}
+
+	_, err := a.crdClient.CrdV1alpha2().IPPools().UpdateStatus(context.TODO(), newPool, metav1.UpdateOptions{})
+	if err != nil {
+		klog.Warningf("IP Pool %s update with status %+v failed: %+v", newPool.Name, newPool.Status, err)
+		return err
+	}
+	klog.InfoS("IP Pool update successful", "pool", newPool.Name, "allocation", newPool.Status)
+	return nil
+
+}
+
+// Update pool status to delete released IP allocation, and keep preallocation information
 func (a *IPPoolAllocator) removePoolUsage(ipPool *v1alpha2.IPPool, ip net.IP) error {
 
 	ipString := ip.String()
 	newPool := ipPool.DeepCopy()
 	var newList []v1alpha2.IPAddressState
-	for _, entry := range ipPool.Status.IPAddresses {
+	allocated := false
+	for i := range ipPool.Status.IPAddresses {
+		entry := *ipPool.Status.IPAddresses[i].DeepCopy()
```

Comment:
Why it needs to make a deep copy when the IP doesn't match?

Reply:
Moved `DeepCopy` to `else` section.